apps/docs/lib/userAuth.ts

```diff
@@ -0,0 +1,58 @@
+import { Session } from '@supabase/supabase-js'
+import { gotrueClient } from 'common'
+import { LOCAL_STORAGE_KEYS, remove } from './storage'
+import { useEffect } from 'react'
+
+export const auth = gotrueClient
+
+let currentSession: Session | null
+
+auth.onAuthStateChange((event, session) => {
+  currentSession = session
+
+  if (event === 'SIGNED_OUT') {
+    Object.keys(LOCAL_STORAGE_KEYS).forEach((key) => {
+      remove('local', LOCAL_STORAGE_KEYS[key])
+    })
+  }
```

Comment:
i think the client library automatically clears the session from local storage on sign out

reference: https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts#L1493

Reply:
good to know! 👍🏼 I still need to clear out the data that the frontend saves though